[Proposal] Add `/run_checks` permission

sushichop opened the pull request:
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
* [x] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [ ] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

### Pull Request Description

I'd like to limit the users who can execute `/run_checks` to collaborator and admin (I think it was the expected behavior?). I also simplified the action for checking out the pull requests from the fork. In fact, I have already implemented this method in some projects and they works fine.

Do you prefer this proposal? TBH, I'm not sure we really need this feature(limit the users who can execute `/run_checks`), but we can do it😅 and it gets better, I think.

As a reminder,  this change will work fine after the merge(not at pull request) due to GitHub Action issue_event specification.

Thanks.

sushichop commented:
/run_checks

ffried commented on .github/workflows/validations.yml:
```diff
@@ -39,29 +39,13 @@ jobs:
         with:
-          repository: ${{fromJSON(steps.pr-details.outputs.result).head.repo.full_name}}
-          ref: ${{fromJSON(steps.pr-details.outputs.result).head.ref}}
+          ref: refs/pull/${{ github.event.issue.number }}/head
```
I'm not sure that this will work in all situations. I've manually fetched the PR to be independent of any forking situation (e.g. what about 2+ level forks coming back with a PR into upstream).

sushichop replied:
I see. I certainly hadn't thought of that case... So I'd like to revert these line(check out the PR form the fork)

sushichop commented:
@ffried 
Thanks!  I'm glad to hear your idea! I reverted the lines about checking out the PR from the fork.
Now I've just added execution(`run_check`) limitation. If it's OK, could you merge this?

sushichop commented:
/run_checks (just in case🙂)

ffried commented:
@sushichop After a quick glance at your GitHub action, I saw that you can post a comment if the user does not have the required permissions. Will this happen by default? If not, would you mind adding this to the action? I think that'll help to understand why the checks do not run in certain cases.

sushichop commented:
@ffried 
Oh, I'm very glad you have checked my actions! It doesn't comment on anything by default.
So I added action definition when a user is not permitted.

ffried commented:
/run_checks
